eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/model/eva/SubmittedVariantEntity.java

```diff
@@ -107,13 +109,39 @@ public SubmittedVariantEntity(Long accession, String hashedMessage, String refer
         setValidated(validated);
     }
 
+    /**
+     * This constructor should only be used when the mapping weight is required
+     */
+    public SubmittedVariantEntity(Long accession, String hashedMessage, String referenceSequenceAccession,
+                                  int taxonomyAccession, String projectAccession, String contig, long start,
+                                  String referenceAllele, String alternateAllele, Long clusteredVariantAccession,
+                                  Boolean supportedByEvidence, Boolean assemblyMatch, Boolean allelesMatch,
+                                  Boolean validated, int version, Integer mapWeight) {
+        super(hashedMessage, accession, version);
+        this.referenceSequenceAccession = referenceSequenceAccession;
+        this.taxonomyAccession = taxonomyAccession;
+        this.projectAccession = projectAccession;
+        this.contig = contig;
+        this.start = start;
+        this.referenceAllele = referenceAllele;
+        this.alternateAllele = alternateAllele;
+        this.clusteredVariantAccession = clusteredVariantAccession;
+
+        setSupportedByEvidence(supportedByEvidence);
+        setAssemblyMatch(assemblyMatch);
+        setAllelesMatch(allelesMatch);
+        setValidated(validated);
```

Comment:
Could we invoke the previous constructor in lieu of all the above statements?

Reply:
Done